src/Raven.Server/Rachis/Follower.cs

```diff
@@ -120,13 +117,46 @@ private void FollowerSteadyState()
                             var task = Concurrent_SendAppendEntriesPendingToLeaderAsync(cts, _term, lastLogIndex);
                             try
                             {
-                                bool hasRemovedFromTopology;
+                                bool hasRemovedFromTopology = false;
 
-                                (hasRemovedFromTopology, lastLogIndex, lastTruncate, lastCommit) = ApplyLeaderStateToLocalState(sp,
-                                    context,
-                                    entries,
-                                    appendEntries);
+                                try
+                                {
+                                    (hasRemovedFromTopology, lastLogIndex, lastTruncate, lastCommit) = ApplyLeaderStateToLocalState(sp,
+                                        context,
+                                        entries,
+                                        appendEntries);
 
+                                    entries.Clear();
+                                }
+                                catch
+                                {
+                                    var processed = 0;
+                                    try
+                                    {
+                                        foreach (var entry in entries)
+                                        {
+                                            (hasRemovedFromTopology, lastLogIndex, lastTruncate, lastCommit) = ApplyLeaderStateToLocalState(sp,
+                                                context,
+                                                new List<RachisEntry>
+                                                {
+                                                    entry
+                                                },
+                                                appendEntries);
+
+                                            if (hasRemovedFromTopology)
+                                            {
+                                                break;
+                                            }
+
+                                            processed++;
+                                        }
+                                    }
+                                    finally
+                                    {
+                                        entries.RemoveRange(0, processed);
```

Comment:
We'll have entries from previous reads, then try to re-process them?
But we wait for the next network read?